include/pybind11/eigen.h

```diff
@@ -346,7 +346,8 @@ struct type_caster<Type, enable_if_t<is_eigen_dense_plain<Type>::value>> {
 
     operator Type*() { return &value; }
     operator Type&() { return value; }
-    operator Type&&() && { return std::move(value); }
+    operator Type&&() { return std::move(value); }
+    operator Type() { return std::move(value); }
```

Comment:
Does `operator Type()` here need the SFINAE treatement?

Reply:
The Eigen types always are move-constructible. Hence, no need for protection, I think.